src/config/media.config.ts

```diff
@@ -44,18 +45,29 @@ const mediaSchema = Joi.object({
         otherwise: Joi.optional(),
       }).label('HD_MEDIA_BACKEND_FILESYSTEM_UPLOAD_PATH'),
     },
-    s3: Joi.when('...use', {
+    s3: Joi.when('use', {
       is: Joi.valid(BackendType.S3),
       then: Joi.object({
         accessKey: Joi.string().label('HD_MEDIA_BACKEND_S3_ACCESS_KEY'),
         secretKey: Joi.string().label('HD_MEDIA_BACKEND_S3_SECRET_KEY'),
-        endPoint: Joi.string().label('HD_MEDIA_BACKEND_S3_ENDPOINT'),
-        secure: Joi.boolean().label('HD_MEDIA_BACKEND_S3_SECURE'),
-        port: Joi.number().label('HD_MEDIA_BACKEND_S3_PORT'),
+        bucket: Joi.string().label('HD_MEDIA_BACKEND_S3_BUCKET'),
+        endPoint: Joi.string()
+          .pattern(/^(?!http).*/)
+          .label('HD_MEDIA_BACKEND_S3_ENDPOINT')
+          .messages({
+            'string.pattern.base':
+              '{#label} can not start with "http" or "https"',
+          }),
```

Comment:
Is this required to be an IP address?

Reply:
No it can be a normal hostname, too. But the lib want's the assembly the protocol itself. And I think it's better to let the user break it down instead of trying to get the appropriate parts from a single url extracted…

Reply:
And what about my hostname `http-balancer.example.com`? That would be blocked with your pattern.

Reply:
I therefore suggest to check for the colon after the http/https as well.

```
.pattern(/^(?!https?:).*/)
```

Reply:
that's a good point. I'll make an appropriate change

Reply:
I think we could just throw the URL into a URL object? That should resolve the URL parts for us, without the user need to configure multiple things. Does that sound like an idea?

Reply:
I changed it, tell me what you think about it this way.